include/deal.II/distributed/shared_tria.h

```diff
@@ -23,6 +23,8 @@
 #include <deal.II/base/smartpointer.h>
 #include <deal.II/base/template_constraints.h>
 #include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/filtered_iterator.h>
```

Comment:
I don't think you need these include files in the `.h` file. Move the `#include` directives into the `.cc` file as necessary.